spotbugs/src/main/java/edu/umd/cs/findbugs/detect/PublicAttributes.java

```diff
@@ -0,0 +1,224 @@
+                "finalize".equals(methodName) ||
+                "this".equals(methodName) ||
+                "_jspInit".equals(methodName) ||
+                "_jspDestroy".equals(methodName);
+    }
+
+    private static @CheckReturnValue Field getClassField(XField field, String dottedClassName) {
+        try {
+            JavaClass cls = Repository.lookupClass(dottedClassName);
+            for (Field clsField : cls.getFields()) {
+                if (field.getName().equals(clsField.getName())) {
+                    return clsField;
+                }
+            }
+        } catch (ClassNotFoundException cnfe) {
+            AnalysisContext.reportMissingClass(cnfe);
+        }
+        return null;
+    }
+
+    private static boolean isImmutable(String dottedClassName) {
+        if ("java.lang.String".equals(dottedClassName) ||
+                "java.lang.Character".equals(dottedClassName) ||
+                "java.lang.Byte".equals(dottedClassName) ||
+                "java.lang.Short".equals(dottedClassName) ||
+                "java.lang.Integer".equals(dottedClassName) ||
+                "java.lang.Long".equals(dottedClassName) ||
+                "java.lang.Float".equals(dottedClassName) ||
+                "java.lang.Double".equals(dottedClassName) ||
+                "java.lang.Boolean".equals(dottedClassName)) {
+            return true;
+        }
+
+        try {
+            JavaClass cls = Repository.lookupClass(dottedClassName);
+
+            return isImmutable(cls);
+        } catch (ClassNotFoundException cnfe) {
+            AnalysisContext.reportMissingClass(cnfe);
+        }
+        return false;
+    }
+
+    private static boolean isImmutable(JavaClass cls) {
+        if (!cls.isFinal()) {
+            return false;
+        }
+
+        for (Field field : cls.getFields()) {
+            if (!field.isStatic() &&
+                    (!field.isFinal() || !field.isPrivate())) {
+                return false;
+            }
+        }
+
+        try {
+            JavaClass sup = cls.getSuperClass();
+            return isImmutable(sup);
+        } catch (ClassNotFoundException cnfe) {
+            return true;
+        }
+    }
+
+    private static boolean looksLikeASetter(String methodName) {
+        return methodName.startsWith("set") ||
+                methodName.startsWith("put") ||
+                methodName.startsWith("add") ||
+                methodName.startsWith("insert") ||
+                methodName.startsWith("delete") ||
+                methodName.startsWith("remove") ||
+                methodName.startsWith("erase") ||
+                methodName.startsWith("clear") ||
+                methodName.startsWith("push") ||
+                methodName.startsWith("pop") ||
+                methodName.startsWith("enqueue") ||
+                methodName.startsWith("dequeue") ||
+                methodName.startsWith("write") ||
+                methodName.startsWith("append");
```

Comment:
Maybe here too... Although this is trickier because you'd have to use some lambda "magic".